opencog/query/PatternMatchCallback.h

```diff
@@ -238,6 +238,16 @@ class PatternMatchCallback
 		                       const std::map<Handle, Handle> &term_soln) = 0;
 
 		/**
+		 * Called when we are comparing unordered links. It would be useful
+		 * if we want to skip this grounding and move on to the next right away
+		 * without having to compare all of its possible permutations.
+		 */
```

Comment:
There is an easier way to do this, without having to alter the pattern matcher.

What you do in the code below is to always declare a mis-compare, whenever you encounter an unordered link (you return false, when skipping permutation compares)  

You can accomplish the same thing by changing  FuzzyMatcher::link_match() to return false whenever the link type is unordered.